run_linux.go

```diff
@@ -2391,6 +2393,15 @@ func (b *Builder) runSetupRunMounts(mounts []string, secrets map[string]string,
 				// Count is needed as the default destination of the ssh sock inside the container is  /run/buildkit/ssh_agent.{i}
 				sshCount++
 			}
+		case "bind":
+			mount, err := b.getBindMount(tokens, contextDir, rootUID, rootGID, processUID, processGID)
+			if err != nil {
+				return nil, nil, err
+			}
+			if mount != nil {
```

Comment:
Is this check necessary?  Are there cases where `err` is `nil` and `mount` is also `nil`?

Reply:
@nalind since its a pointer  i think it would be ok to do a `nil` before performing any `dereferencing`. Better than any SIGSEGV in future. But i am cool dropping if its a bother.

Reply:
`getSecretMount()` and `getSSHMount()` can return a `nil` mount without also returning an error but `getBindMount()` doesn't.